tests/aggregation_regress/tests.py

```diff
@@ -1318,19 +1318,43 @@ def assertQuerysetResults(queryset):
         queryset = Book.objects.select_related('contact').annotate(num_authors=Count('authors'))
         # Unmanaged origin model.
         with mock.patch.object(Book._meta, 'managed', False):
+            # When it is view. This is autodetected by default, we override the
+            # cached value for tests.
+            with mock.patch.object(Book._meta, '_is_ordinary_table', False, create=True):
+                _, _, grouping = queryset.query.get_compiler(using='default').pre_sql_setup()
+                self.assertEqual(len(grouping), len(Book._meta.fields) + 1)
+                for index, field in enumerate(Book._meta.fields):
+                    self.assertIn(field.name, grouping[index][0])
+                self.assertIn(Author._meta.pk.name, grouping[-1][0])
+                assertQuerysetResults(queryset)
+
+            # When it is a normal unmanaged table
             _, _, grouping = queryset.query.get_compiler(using='default').pre_sql_setup()
-            self.assertEqual(len(grouping), len(Book._meta.fields) + 1)
-            for index, field in enumerate(Book._meta.fields):
-                self.assertIn(field.name, grouping[index][0])
-            self.assertIn(Author._meta.pk.name, grouping[-1][0])
+            self.assertEqual(len(grouping), 1 if connection.features.allows_group_by_pk else 2)
+            self.assertIn('id', grouping[0][0])
+            self.assertNotIn('name', grouping[0][0])
+            self.assertNotIn('contact', grouping[0][0])
```

Comment:
`grouping[0][0]` is a name of the first column, so these two assertions are unnecessary:
```python
self.assertNotIn('name', grouping[0][0])
self.assertNotIn('contact', grouping[0][0]) 
```